system/signal.go

```diff
@@ -0,0 +1,37 @@
+package system
```

Comment:
Graceful shutdown
![image](https://user-images.githubusercontent.com/62853952/101800987-90d30580-3b16-11eb-9ff3-34e4ecd9ec8c.png)

Reply:
Nice... Is this totally new functionality unrelated to the wrapping aspects?